### PR TITLE
Add the ddump-dir option/config value #4225

### LINKED ISSUE #4225: Can't display -ddump-splices result.
```shell
$ stack build --ghc-options='-ddump-splices'
```

I want to display on the standard output, not to save to a file.

### Question

Why does the [extraBuildOptions](https://github.com/commercialhaskell/stack/blob/master/src/Stack/Build/Execute.hs#L1948) function specify the `" -ddump-hi -ddump-to-file"` option by default?

```hs
extraBuildOptions wc bopts = do
    colorOpt <- appropriateGhcColorFlag
    let ddumpOpts = " -ddump-hi -ddump-to-file"
        optsFlag = compilerOptionsCabalFlag wc
        baseOpts = ddumpOpts ++ maybe "" (" " ++) colorOpt
    if toCoverage (boptsTestOpts bopts)
      then do
        hpcIndexDir <- toFilePathNoTrailingSep <$> hpcRelativeDir
        return [optsFlag, "-hpcdir " ++ hpcIndexDir ++ baseOpts]
      else
        return [optsFlag, baseOpts]
```

If remove this option, I think it will work as expected.

### Stack version

```
$ stack --version
Version 1.10.0, Git revision 3b61926935c28efd10c8e6870f927a7d8eb5523c (dirty) (6119 commits) x86_64 hpack-0.29.6
```

### PR DESCRIPTION
**Pull request by waddlaw**
Fixes #4225

Note: Documentation fixes for https://docs.haskellstack.org/en/stable/ should target the "stable" branch, not master.

Please include the following checklist in your PR:

* [x] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by waddlaw** [comment]
@snoyberg this should be all set now :)
Is this alright? Could you please confirm it?

File: src/Stack/Build/Execute.hs [hunk 1]
```diff
@@ -1532,6 +1533,28 @@ singleBuild ac@ActionContext {..} ee@ExecuteEnv {..} task@Task {..} installedMap
                 _ -> return ()
             when hasLibrary $ cabal KeepTHLoading ["register"]
 
+        -- copy ddump-* files
+        let enableDdumpDir = isJust $ boptsDdumpDir eeBuildOpts
+            ddumpPath      = maybe "" T.unpack $ boptsDdumpDir eeBuildOpts
+        when (buildingFinals && enableDdumpDir && not (null ddumpPath)) $ do
```
Comment (snoyberg) [inline src/Stack/Build/Execute.hs]: It looks like a pattern match would be easier to read here, e.g.:

```haskell
case T.unpack <$> boptsDdumpDir eebuildOpts of
  Just ddumpPath | buildingFinals && enableDdumpDir && not (null ddumpPath) -> do ...
  _ -> pure ()
```

Comment (waddlaw) [reply src/Stack/Build/Execute.hs]: That makes sense. That's true :)

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**